### PR TITLE
Move settings json from IQMClients constructor to submit_circuit

### PR DESCRIPTION
**Pull request by LeonWubben**
See https://github.com/iqm-finland/iqm-client/pull/31

### CODE REVIEW

File: docs/user_guide.rst [hunk 1]
```diff
@@ -235,10 +235,14 @@ can be used to specify this correspondence.
...
 
+    import json
+
+
     qubit_mapping = {'Alice': 'QB1', 'Bob': 'QB3'}
 
     with open(iqm_settings_path, 'r') as f:
-        sampler = IQMSampler(iqm_server_url, f.read(), adonis, qubit_mapping=qubit_mapping)
+        settings = json.loads(f.read())
+        sampler = IQMSampler(iqm_server_url, adonis, qubit_mapping=qubit_mapping, settings=settings)
```
Comment (smite) [inline docs/user_guide.rst:246]:
```suggestion

    sampler = IQMSampler(iqm_server_url, adonis, qubit_mapping=qubit_mapping, settings=settings)
```

File: docs/user_guide.rst [hunk 2]
```diff
@@ -235,10 +235,14 @@ can be used to specify this correspondence.
 
 .. code-block:: python
 
+    import json
+
+
     qubit_mapping = {'Alice': 'QB1', 'Bob': 'QB3'}
 
     with open(iqm_settings_path, 'r') as f:
```
Comment (smite) [inline docs/user_guide.rst:243]:
```suggestion
    with open(iqm_settings_path, 'r', encoding='utf-8') as f:
```

Comment (smite) [reply docs/user_guide.rst:243]: I think pylint recommends explicit encondings.

File: src/cirq_iqm/iqm_sampler.py [hunk 3]
```diff
@@ -149,12 +119,15 @@ def run_sweep(
     def _send_circuits(
             self,
             circuits: list[cirq.Circuit],
-            repetitions: int = 1,
+            qubit_mapping: dict[str, str],
```
Comment (smite) [inline src/cirq_iqm/iqm_sampler.py:122]:
```suggestion
            qubit_mapping: Optional[dict[str, str]],
```
It can still be None at this point, right?

Comment (hay-k) [reply src/cirq_iqm/iqm_sampler.py:122]: yes

**Review by smite — APPROVED** [review]

**Pull request merged**